contracts/5/SushiswapGoblin.sol

```diff
@@ -0,0 +1,250 @@
+        path[0] = address(sushi);
+        path[1] = address(weth);
+        router.swapExactTokensForETH(reward.sub(bounty), 0, path, address(this), now);
+        // 4. Use add ETH strategy to convert all ETH to LP tokens.
+        addStrat.execute.value(address(this).balance)(address(0), 0, abi.encode(fToken, 0));
+        // 5. Mint more LP tokens and stake them for more rewards.
+        masterChef.deposit(pid, lpToken.balanceOf(address(this)));
+        emit Reinvest(msg.sender, reward, bounty);
+    }
+
+    /// @dev Work on the given position. Must be called by the operator.
+    /// @param id The position ID to work on.
+    /// @param user The original user that is interacting with the operator.
+    /// @param debt The amount of user debt to help the strategy make decisions.
+    /// @param data The encoded data, consisting of strategy address and calldata.
+    function work(uint256 id, address user, uint256 debt, bytes calldata data) 
+        external payable 
+        onlyOperator nonReentrant 
+    {
+        // 1. Convert this position back to LP tokens.
+        _removeShare(id);
+        // 2. Perform the worker strategy; sending LP tokens + ETH; expecting LP tokens + ETH.
+        (address strat, bytes memory ext) = abi.decode(data, (address, bytes));
+        require(okStrats[strat], "unapproved work strategy");
+        lpToken.transfer(strat, lpToken.balanceOf(address(this)));
+        Strategy(strat).execute.value(msg.value)(user, debt, ext);
+        // 3. Add LP tokens back to the farming pool.
+        _addShare(id);
+        // 4. Return any remaining ETH back to the operator.
+        SafeToken.safeTransferETH(msg.sender, address(this).balance);
+    }
+
+    /// @dev Return maximum output given the input amount and the status of Uniswap reserves.
+    /// @param aIn The amount of asset to market sell.
+    /// @param rIn the amount of asset in reserve for input.
+    /// @param rOut The amount of asset in reserve for output.
+    function getMktSellAmount(uint256 aIn, uint256 rIn, uint256 rOut) public pure returns (uint256) {
+        if (aIn == 0) return 0;
+        require(rIn > 0 && rOut > 0, "bad reserve values");
+        uint256 aInWithFee = aIn.mul(997);
+        uint256 numerator = aInWithFee.mul(rOut);
+        uint256 denominator = rIn.mul(1000).add(aInWithFee);
+        return numerator / denominator;
+    }
+
+    /// @dev Return the amount of ETH to receive if we are to liquidate the given position.
+    /// @param id The position ID to perform health check.
+    function health(uint256 id) external view returns (uint256) {
+        // 1. Get the position's LP balance and LP total supply.
+        uint256 lpBalance = shareToBalance(shares[id]);
+        uint256 lpSupply = lpToken.totalSupply(); // Ignore pending mintFee as it is insignificant
+        // 2. Get the pool's total supply of WETH and farming token.
+        uint256 totalWETH = weth.balanceOf(address(lpToken));
```

Comment:
Please update to the latest goblin version (use lpToken.reserves())

Reply:
Updated to `lpToken.reserves()`